### ISSUE TITLE
Feature Request: smoother process for SharePoint Shared Library sync

### ISSUE THREAD

**Issue by jdpipe** [abraunegg/onedrive#2474]
### Is your feature request related to a problem? Please describe.

I went through [the instructions for SharePoint Shared Library Sync](https://github.com/abraunegg/onedrive/blob/master/docs/SharePoint-Shared-Libraries.md), in particular using the [Ubuntu Snap version of OneDrive for Linux](https://snapcraft.io/onedrive-cli).

Eventually it worked, but only once I realised that I have to use specific folders withing `~/snap/onedrive-cli/current` for it to work (more details [here](https://github.com/boukendesho/onedrive-cli-snap/issues/1)). To avoid this issue, perhaps it would be helpful if the OneDrive for Linux client could extend its normal internal configuration mechanism so that extra config folders are not required. Perhaps some Snap-specific code could be considered to make this process simpler.

### Describe the solution you'd like

I think that Snap is one of the few ways to get the latest version of `onedrive-cli` up and running without a big compilation/setup, so it would be nice some effort can be made to make `onedrive-cli` work a little more smoothly in this context. Perhaps a user instruction suggestion a symlink for more convenient access to their files in `~/snap/onedrive-cli/common/PROJECTNAME` for example. Perhaps it is possible for `onedrive-cli` to detect when it is running in a Snap context, and alter its behaviour slightly.

Two suggestions:
1. when setting up a SharePoint Shared Library Sync, support `onedrive-cli` to create its own configuration files/folders, so that the user doesn't need to do it. This needs more automation of the current process.
2. when running within a Snap, give the user information about setting up a Symlink to the created folder, since otherwise they may struggle to find where their files went.

### Describe alternatives you've considered

I managed to get something work, but it was fairly confusing, because of the limitations of where Snaps can store files. Perhaps there is an easier way, eg by granting access to the Snap to the folder(s) it needs.

**Comment by abraunegg** [comment]
@jdpipe 



> To avoid this issue, perhaps it would be helpful if the OneDrive for Linux client could extend its normal internal configuration mechanism so that extra config folders are not required

Unfortunately this is what is required - for each 'OneDrive' configuration, a separate configuration path to store that data is needed. 

> I think that Snap is one of the few ways to get the latest version of onedrive-cli up and running without a big compilation/setup,

This is not correct. The client is bundled as a package for almost all Linux distributions:
* https://repology.org/project/onedrive/versions

Try using the correct package for your distribution rather than relying on a Snap.

> when setting up a SharePoint Shared Library Sync, support onedrive-cli to create its own configuration files/folders, so that the user doesn't need to do it. This needs more automation of the current process.

It already supports putting the data in it's own location - all you need to do is specify the `--confdir` option and this will be used to store the data. 

Additionally as an FYI - I have nothing to do with any 'onedrive' Snap. Issues or problems with it should be raised to that maintainer. If the Snap is doing weird things with where your data is going - that is not this clients problem - sorry.

**Comment by jdpipe** [comment]
Hi @abrauengg
Unfortunately I was using an Ubuntu 18.04 system for this, for which only a very old version of OneDrive is available -- without the SharePoint Shared Library feature that I was interested in. I had to use the Snap.

The issue with Snap is that it limits the places that an app can save its files. All of the files have to be stored within `~/snap/onedrive-cli/...` as far as I can see (maybe there are ways to loosen the constraints somehow).

But thanks very much, once again, for implementing the SharePoint Shared Library sync. It's a bit help, and I'm sure it will get more user-friendly over time. Great project!

**Comment by abraunegg** [comment]
@jdpipe 
Ubuntu 18.04 reached end of life on May 31, 2023 and is no longer a supported platform. You should not be using Ubuntu 18.x anymore and should strongly consider migrating your system to a current Canocial supported platform.

The nuances on how Snap implements filesystem access I also have zero control over. Please have a read of this:

https://forum.snapcraft.io/t/request-for-classic-confinement-for-onedrive/18984

**Comment by jdpipe** [comment]
Thanks @abraunegg 
I guess one small thing that would have been helpful here would have been an error saying "because you are using a Snap version of OneDrive, you can't write to files outside the `~/snap` directory. Hence you need to put your config file and sync folder somewhere in there." But I get that this is a hassle and probably most people are not using a version of OneDrive that is limited by such packaging constraints. The error messages were quite puzzling -- I was trying all sorts of `chmod` things thinking it was some issue like you get when your `~/.ssh` folder settings are wrong.
As for 18.04, it still has "extended support" via Ubuntu Pro, which is free for personal users FYI. But yes, I should upgrade/rebuild this machine.

**Comment by abraunegg** [comment]
@jdpipe 
The thing to note here - I have nothing to do with any Snap - so any use or error message or anything that is surfaced by using a Snap - is out of my realm of control.

**Comment by abraunegg** [comment]
This issue has been automatically locked since there has not been any recent activity after it was closed. Please open a new issue for related bugs.
